TACC/Core-Portal
# task/WP-43: System Monitor UI Update

## Overview

The goal for this PR was to update the System Monitor functionality and add an additional page that displays detailed queue information for a particular system. This new page can be accessed via the main sidebar or by clicking on a system name in the existing Dashboard System Monitor

## Related

* [WP-43](https://jira.tacc.utexas.edu/browse/WP-43)

## Changes

- Added new System Status page which goes to the detailed information of the first system by default. Clicking on the systems will show detailed queue information for each system 
- Added ability to click system names in System Monitor that redirect to the System Status page with detailed queue information 
- Added `system_name` parameter to `SysmonDataView` endpoint. When the `system_name` parameter is sent a call is made to the new [TAP](https://tap.tacc.utexas.edu/status/maverick2.tacc.utexas.edu/) endpoint to get the systems' queue information


## Testing

1. Log onto the portal and go to the main Dashboard
2. Navigate to the System Status page by clicking on the System Status sidebar item
3. Ensure information for first system in list is shown by default
4. Click on a system in the sidebar and ensure that queue information is shown 
5. Go back to Dashboard and click on a system from the System Monitor widget
6. Ensure that page is redirected to the System Status page with the correct systems information being displayed
7. Click on View All Systems button and ensure page is redirected to default System Monitor view

## UI

<img width="2046" alt="Screen Shot 2023-06-08 at 6 52 18 PM" src="https://github.com/TACC/Core-Portal/assets/23081932/28ca786b-34b0-4e76-9eae-5bedd6d10101">


## Notes
TODO: 

- [x] Write tests for new System Status components created 
- [x] Update System Monitor tests and test the new functionality added
